src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -2,23 +2,207 @@
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using System.Linq.Expressions;
 using System.Text.RegularExpressions;
 using GeneratorV2.Data;
 using GeneratorV2.Writing;
 
 namespace GeneratorV2.Process
 {
+    public sealed class VectorOverloader : IOverloader
+    {
+        private static readonly Regex VectorNameMatch = new Regex("([1-4])([f|d|h|i])v", RegexOptions.Compiled);
+        private readonly HashSet<string> _existingVectorTypes = new HashSet<string>()
+        {
+            "Vector2",
+            "Vector2d",
+            "Vector2h",
+            "Vector2i",
+            "Vector3",
+            "Vector3d",
+            "Vector3h",
+            "Vector3i",
+            "Vector4",
+            "Vector4d",
+            "Vector4h",
+            "Vector4i",
+        };
+
+        public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List<Overload>? newOverloads)
+        {
+            Match m = VectorNameMatch.Match(overload.OverloadName);
+            if (!m.Success || !int.TryParse(m.Groups[1].Value, out int vectorSize))
```

Comment:
I try to avoid `!` as it's very easy to miss. I would write `m.Success == false` to make it more explicit that you are looking for the negative case.

We also don't need the `int.TryParse`, if the regex matched the int will be parsable so we can just do `int.Parse` and not have that in the if-statement.